Array format support is not reflected in the type definition for typescript

**Describe the bug**
Array format is supported by #906, but typescript gives the following type error.

```typescript
dayjs('2020-02-02', ['YYYY', 'YYYY-DD-MM'])
```

```
 error TS2345: Argument of type 'string[]' is not assignable to parameter of type 'string | { locale?: string | undefined; format?: string | undefined; utc?: boolean | undefined; } | undefined'.
      Type 'string[]' is not assignable to type 'string'.
```

I think we need to update the function signature for typescript.

**Expected behavior**
It works just as well as javascript

**Information**
 - Day.js Version : 1.8.28
 - OS: macOS

--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -6,7 +6,7 @@ declare function dayjs (date?: dayjs.ConfigType, option?: dayjs.OptionType, loca
 declare namespace dayjs {
   export type ConfigType = string | number | Date | Dayjs
 
-  export type OptionType = { locale?: string, format?: string, utc?: boolean } | string
+  export type OptionType = { locale?: string, format?: string, utc?: boolean } | string | string[]
 
   type UnitTypeShort = 'd' | 'M' | 'y' | 'h' | 'm' | 's' | 'ms'
   export type UnitType = 'millisecond' | 'second' | 'minute' | 'hour' | 'day' | 'month' | 'year' | 'date' | UnitTypeShort;

